src/plugins/plugin.filler.js

```diff
@@ -250,11 +291,45 @@ function doFill(ctx, points, mapper, view, color, loop) {
 		}
 	}
 
-	drawArea(ctx, curve0, curve1, len0, len1);
+	sets = drawArea(ctx, curve0, curve1, len0, len1, mode, area);
+	if (sets) {
+		clipAreaPointsSets.push(sets[0]);
+		fillAreaPointsSets.push(sets[1]);
+	}
+
+	if (mode !== 'all') {
+		for (i = 0; i < clipAreaPointsSets.length; i++) {
+			set = clipAreaPointsSets[i];
+			jlen = set.length;
+			// Have edge lines straight
+			ctx.moveTo(set[0].x, set[0].y);
+			ctx.lineTo(set[1].x, set[1].y);
+			for (j = 2; j < jlen - 1; j++) {
```

Comment:
consistency (and one line less): `for (j = 2, jlen  = set.length; j < jlen - 1; ++j)`